core/storage/suggestion/gae_models.py

```diff
@@ -0,0 +1,262 @@
+SUGGESTION_EDIT_STATE_CONTENT = 'edit_exploration_state_content'
+
+SUGGESTION_TYPE_CHOICES = [
+    SUGGESTION_EDIT_STATE_CONTENT
+]
+
+# Defines what is the minimum role required to review suggestions
+# of a particular type.
+SUGGESTION_MINIMUM_ROLE_FOR_REVIEW = {
+    SUGGESTION_EDIT_STATE_CONTENT: feconf.ROLE_ID_EXPLORATION_EDITOR
+}
+
+# Constants defining various contribution types.
+SCORE_TYPE_CONTENT = 'content'
+SCORE_TYPE_TRANSLATION = 'translation'
+
+SCORE_TYPE_CHOICES = [
+    SCORE_TYPE_CONTENT,
+    SCORE_TYPE_TRANSLATION
+]
+
+# The delimiter to be used in the suggestion ID.
+SUGGESTION_ID_DELIMITER = '.'
+# The delimiter to be used in score category field.
+SCORE_CATEGORY_DELIMITER = '.'
+
+
+class SuggestionModel(base_models.BaseModel):
+    """Model to store suggestions made by Oppia users.
+
+    The ID of the suggestions are created by joining target_type, target_id and
+    thread_instance_ID using the SUGGESTION_ID_DELIMITER.
+    """
+
+    # The type of suggestion.
+    suggestion_type = ndb.StringProperty(
+        required=True, indexed=True, choices=SUGGESTION_TYPE_CHOICES)
+    # The type of the target entity which the suggestion is linked to.
+    target_type = ndb.StringProperty(
+        required=True, indexed=True, choices=TARGET_TYPE_CHOICES)
+    # The ID of the target entity being suggested to.
+    target_id = ndb.StringProperty(required=True)
+    # The version number of the target entity at the time of creation of the
+    # suggestion.
+    target_version_at_submission = ndb.IntegerProperty(required=True)
+    # Status of the suggestion.
+    status = ndb.StringProperty(
+        required=True, indexed=True, choices=STATUS_CHOICES)
+    # The ID of the author of the suggestion.
+    author_id = ndb.StringProperty(required=True, indexed=True)
+    # The ID of the reviewer assigned to review the suggestion.
+    assigned_reviewer_id = ndb.StringProperty(required=False, indexed=True)
+    # The ID of the reviewer who accepted the suggestion.
+    reviewer_id = ndb.StringProperty(required=False, indexed=True)
+    # The change command linked to the suggestion. Contains the details of the
+    # change.
+    change_cmd = ndb.JsonProperty(required=True, indexed=False)
+    # The category to score the suggestor in. This field will contain 2 values
+    # separated by a ., the first will be a value from SCORE_TYPE_CHOICES and
+    # the second will be the subcategory of the suggestion.
+    score_category = ndb.StringProperty(required=True, indexed=True)
+
+    @classmethod
+    def get_instance_id(
+            cls, target_type, target_id, thread_instance_id):
+        """Concatenates various parameters and gives the ID of the suggestion
+        model.
+
+        Args:
+            target_type: str. The type of target being edited.
+            target_id: str. The ID of the target being edited.
+            thread_instance_id: str. The instance ID of the feedback thread
```

Comment:
thread_id will do, and you can just say "The ID of the feedback thread" in the same way that you say "The ID of the target".

Reply:
Actually I might be a bit confused here. I thought you were going to just have the IDs for the thread and the suggestion match up directly? So do you actually need the target type and target ID at all to construct the instance ID?

Reply:
Renamed the parameter to thread_id